### ISSUE TITLE
[prefer-snapshot-hint] is it supposed to warn on multiple snapshots in a different test but the same describe block?

### ISSUE THREAD

**Issue by lostfictions** [jest-community/eslint-plugin-jest#1074]
Hey there, thanks for your quick work fixing #1068! Unfortunately, I'm still seeing an issue with `prefer-snapshot-hint`. 

With `"jest/prefer-snapshot-hint": ["warn", "multi"]` set, this works fine:

```ts
describe("my cool test", () => {
  it("requires hints for a test with multiple snapshots", () => {
    expect(1).toMatchSnapshot("a cool hint");
    expect(2).toMatchSnapshot("another cool hint");
  });
});
```

This also works fine:

```ts
describe("my cool test", () => {
  it("doesn't require hints for a test with a single snapshot", () => {
    expect(2).toMatchSnapshot();
  });
});
```

However, this yields a warning:

```ts
describe("my cool test", () => {
  it("doesn't require hints for a test with a single snapshot", () => {
    expect(2).toMatchSnapshot();
    //        ^ Warning: You should provide a hint for this snapshot
  });

  it("requires hints for a test with multiple snapshots", () => {
    expect(1).toMatchSnapshot("a cool hint");
    expect(2).toMatchSnapshot("another cool hint");
  });
});
```

I see in the [rule description](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/prefer-snapshot-hint.md#multi-default) there's a parenthetical "(meaning it includes nested calls)" -- is this what that means? The behaviour is a bit surprising to me -- I thought "nested calls" meant it would treat the scope as including all functions called from within an `it()` block, not all `it()` blocks within a `describe()` block.

**Comment by G-Rath** [comment]
Yup that's another bug - your thinking on what "nested calls" means is correct - the behaviour shouldn't change just because it gets wrapped in a `describe` like that.
